internal/cmd/gocdk/init.go

```diff
@@ -70,30 +71,61 @@ func init_(ctx context.Context, pctx *processContext, args []string) error {
 		return xerrors.Errorf("gocdk init: %w", err)
 	}
 
-	if err := os.MkdirAll(projectDir, 0777); err != nil {
-		return xerrors.Errorf("gocdk init: %w", err)
-	}
-
 	tmplValues := struct {
 		ProjectName string
 		ModulePath  string
 	}{
 		ProjectName: filepath.Base(projectDir),
 		ModulePath:  modpath,
 	}
-	for fileName, templateSource := range InitTemplates {
-		tmpl := template.Must(template.New(fileName).Parse(templateSource))
-		buf := new(bytes.Buffer)
-		if err := tmpl.Execute(buf, tmplValues); err != nil {
-			return xerrors.Errorf("gocdk init: %w", err)
-		}
+	if err := materializeTemplateDir(projectDir, "init", tmplValues); err != nil {
+		return xerrors.Errorf("gocdk init: %w", err)
+	}
+	return nil
+}
 
-		fullPath := filepath.Join(projectDir, filepath.FromSlash(fileName))
-		if err := os.MkdirAll(filepath.Dir(fullPath), 0777); err != nil {
-			return xerrors.Errorf("gocdk init: %w", err)
+func materializeTemplateDir(dst string, srcRoot string, data interface{}) error {
```

Comment:
Later we might want to move this elsewhere, since IIUC we should be able to use it for `add demo` as well.

Reply:
Ack.